prql-compiler/src/sql/anchor.rs

```diff
@@ -175,7 +175,7 @@ impl AnchorContext {
                     columns = table_def.columns.iter().map(|c| c.id).collect();
                 }
                 Transform::Select(cols) => columns = cols.clone(),
-                Transform::Aggregate(cols) => columns = cols.clone(),
+                Transform::Aggregate { by } => columns = by.clone(),
                 Transform::Join { with, .. } => {
```

Comment:
Aggregate should not be here anymore, because it doesn't have any effects on columns anymore.